zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/container/AdHocSubProcessProcessor.java

```diff
@@ -203,26 +204,66 @@ private void activateElements(
                 stateTransitionBehavior.activateChildInstance(context, elementToActivate));
   }
 
+  @Override
+  public Either<Failure, ?> beforeExecutionPathCompleted(
```

Comment:
Similar to [MultiInstanceBodyProcessor](https://github.com/camunda/camunda/blob/main/zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/container/MultiInstanceBodyProcessor.java), this resolves the completion condition in this before handler. The result is passed to `afterExecutionPathCompleted ` in `satisfiesCompletionCondition`.